src/main/java/org/example/servise/AnimalSerialize.java

```diff
@@ -0,0 +1,52 @@
+package org.example.servise;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.example.model.Animal;
+import org.example.model.Extensions;
+
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import java.util.List;
+
+
+public class AnimalSerialize {
+
+    private final ObjectMapper objectMapper = new JsonMapper();
+
+    public void serialize(List<Animal> animals, File file) {
+        try {
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            objectMapper.writeValue(file, animals);
+
+        } catch (IOException e) {
+            System.out.println("serialization error " + e.getMessage());
```

Comment:
in your case, if you cannot write data to the file, you just print a message to the console. In that kind of scenario method called serialize() will never know that there were some problem with serialization